include/crypto/evp.h

```diff
@@ -178,6 +178,30 @@ extern const EVP_KDF ss_kdf_meth;
 extern const EVP_KDF x963_kdf_meth;
 extern const EVP_KDF x942_kdf_meth;
 
+struct evp_rand_st {
+    OSSL_PROVIDER *prov;
+    int name_id;
+    CRYPTO_REF_COUNT refcnt;
+    CRYPTO_RWLOCK *lock;
+
+    OSSL_OP_rand_newctx_fn *newctx;
+    OSSL_OP_rand_freectx_fn *freectx;
+    OSSL_OP_rand_reseed_fn *reseed;
+    OSSL_OP_rand_get_bytes_fn *get_bytes;
+    OSSL_OP_rand_gettable_params_fn *gettable_params;
+    OSSL_OP_rand_gettable_ctx_params_fn *gettable_ctx_params;
+    OSSL_OP_rand_settable_ctx_params_fn *settable_ctx_params;
+    OSSL_OP_rand_get_params_fn *get_params;
+    OSSL_OP_rand_get_ctx_params_fn *get_ctx_params;
+    OSSL_OP_rand_set_ctx_params_fn *set_ctx_params;
+};
+
+extern const EVP_RAND seed_rand_meth;
+extern const EVP_RAND ctr_drbg_rand_meth;
+extern const EVP_RAND digest_drbg_rand_meth;
+extern const EVP_RAND hmac_drbg_rand_meth;
+extern const EVP_RAND test_rand_meth;
```

Comment:
These aren't needed, are they?

Reply:
Not yet, but they will be once the rest of the methods are added.
I'm fine deleting them for the moment.

Reply:
I'm lost. Why are they needed?

Reply:
They are not needed at the moment.
They will be needed once the DRBGs and seed source are moved across to this framework.

`test_rand_meth` is needed already.


I'm quite happy to delete this and put them back when the rest is done (after the F2F most likely).

Reply:
You don't seem to understand. With fetched algos, *there should be no needed for instances like this*, they are inherently legacy.

Reply:
This is an internal header file which provides access for the providers to declare the implementations.  Where else should they go?

Reply:
That sounds like you're mixing this up with `providers/implementations/include/prov/implementations.h`.

BTW, I just noticed that the legacy KDF stuff hasn't been cleaned away, so that provides an example of what I'm saying:

```
: ; git grep tls1_prf_kdf_meth
include/crypto/evp.h:extern const EVP_KDF tls1_prf_kdf_meth;
: ; 
```

No definition, or use, of `tls1_prf_kdf_meth`

Reply:
I understand now.  The KDF stuff is best left to another PR.  I can fix the rand stuff.

Reply:
I've removed the errant definitions -- they weren't used.

I've removed the KDFs as a separate commit.

Reply:
[g] #10170

Reply:
I removed the commit from here and have approved your commit.